src/NServiceBus.Core/CriticalError/CriticalError.cs

```diff
@@ -49,17 +56,43 @@ public virtual void Raise(string errorMessage, Exception exception)
             Guard.AgainstNull(nameof(exception), exception);
             LogManager.GetLogger("NServiceBus").Fatal(errorMessage, exception);
 
-            if (Endpoint == null)
+            lock (endpointCriticalLock)
```

Comment:
I do think it illustrates the need to refactor how we're doing this entirely, I just don't think the time to do that is now.